x-pack/plugins/rule_registry/common/search_strategy/index.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ValidFeatureId } from '@kbn/rule-data-utils';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { IEsSearchRequest, IEsSearchResponse } from 'src/plugins/data/common';
+import { ParsedTechnicalFields } from '../';
+
+export type RuleRegistrySearchRequest = IEsSearchRequest & {
+  featureIds: ValidFeatureId[];
+  dsl?: estypes.QueryDslQueryContainer;
```

Comment:
My bad here! I did not express my self correctly. I think that we just should do that for simplicity
```suggestion
  query?: { bool: estypes.QueryDslBoolQuery };
```
If we are allowing every type from `QueryDslQueryContainer` I am worried that we might have some unexpected behavior who can override our RBAC on top of the alert. Let's kiss it for now.

Reply:
Sure!